src/lib.rs

```diff
@@ -95,19 +96,260 @@ pub enum FloatingAttachPointType {
+                    sizePercent: percent,
+                },
+            },
+        }
+    }
+
+    fn null_id() -> Clay_ElementId {
+        let inner = MaybeUninit::<Clay_ElementId>::zeroed(); // Creates zero-initialized uninitialized memory
+        unsafe { inner.assume_init() }
+    }
+
+    pub fn sizing_width(&mut self, sizing: Sizing) -> &mut Self {
+        self.inner.sizing.width = Self::sizing(sizing);
+        self
+    }
+
+    pub fn sizing_height(&mut self, sizing: Sizing) -> &mut Self {
+        self.inner.sizing.height = Self::sizing(sizing);
+        self
+    }
+
+    pub fn padding(&mut self, padding: (u16, u16)) -> &mut Self {
+        self.inner.padding.x = padding.0;
+        self.inner.padding.y = padding.1;
+        self
+    } 
+
+    pub fn end(&self) -> TypedConfig {
+        let memory = unsafe { Clay__StoreLayoutConfig(self.inner) };
+
+        TypedConfig {
+            config_memory: memory as _,
+            id: Self::null_id(),
+            config_type: ElementConfigType::Layout as _,
+        }
+    }
+}
+
+pub struct Rectangle {
+    inner: Clay_RectangleElementConfig,
+}
+
+impl Rectangle {
+    pub fn new() -> Self {
+        let inner = MaybeUninit::<Clay_RectangleElementConfig>::zeroed(); // Creates zero-initialized uninitialized memory
+        let inner = unsafe { inner.assume_init() };
+        Self { inner }
+    }
+
+    fn null_id() -> Clay_ElementId {
+        let inner = MaybeUninit::<Clay_ElementId>::zeroed(); // Creates zero-initialized uninitialized memory
+        unsafe { inner.assume_init() }
+    }
+
+    pub fn color(&mut self, color: (f32, f32, f32, f32)) -> &mut Self {
+        self.inner.color = Clay_Color { r: color.0, g: color.1, b: color.2, a: color.3 };
+        self
+    }
+
+    pub fn end(&self) -> TypedConfig {
+        let memory = unsafe { Clay__StoreRectangleElementConfig(self.inner) };
+
+        TypedConfig {
+            config_memory: memory as _,
+            id: Self::null_id(),
+            config_type: ElementConfigType::Rectangle as _,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Clay {
+    // memory owned by clay, destroyed at exit
+    mem_raw: *const u8,
 }
+
+impl Clay {
+    pub fn new(width: f32, height: f32) -> Self {
+        unsafe {
+            let memory_size = Clay_MinMemorySize();
+            let memory = vec![0; memory_size as usize];
+            let mem_raw = Box::into_raw(memory.into_boxed_slice()) as *const u8;
+            let arena = Clay_CreateArenaWithCapacityAndMemory(memory_size as _, mem_raw as _);
+            Clay_Initialize(arena, Clay_Dimensions { width, height });
+
+            Self { mem_raw }
+        }
+    }
+
+    pub fn begin(&self) {
+        unsafe {
+            Clay_BeginLayout();
+        }
+    }
+
+    pub fn end(&self) -> Clay_RenderCommandArray {
+        unsafe {
+            Clay_EndLayout()
+        }
+    }
+
+    pub fn with<F: FnOnce()>(&self, configs: &[TypedConfig], f: F) {
+        unsafe {
```

Comment:
Not a big fan of unsafe block that wrap an entire function body... It allows for unsafe code where it's not always needed, an unsafe block for each unsafe call could be an option, even if it means repeating them quite a bit